### PR TITLE
feat: integrating pino logger

### PR DESCRIPTION
**Pull request by artursudnik**
This PR:
* replaces the standard, customized NestJS logger with a more advanced Pino logger with the ability to track the processing of requests by adding a unique reqestId or using requestId injected with the `X-Request-Id` HTTP request header.
* modifies a message in case of exiting process because of the configuration non-meeting the requirements when starting

### CODE REVIEW

**Review by jrhender — COMMENTED** [review]
@artursudnik rather than replace `logger.service` with `pino-logger`, I'm curious if it would have been possible to wrap `pino-logger` in `logger.service`. It seems to me that this might have been a small change and would have required less refactoring

**Review by hejkerooo — APPROVED** [review]

**Comment by artursudnik** [comment]
@jrhender There is some magic happening in this Pino logger that I don't understand, so I decided to integrate it in a way that is recommended in the documentation to avoid any surprises. It is so flexible that it should work. For example, it is possible to customize the time format and this was the main reason why I added `logger.service` before.

File: authorization-server/src/main.ts [hunk 1]
```diff
@@ -19,17 +20,37 @@ async function bootstrap() {
...
+    SIGHUP: 1,
+    SIGINT: 2,
+    SIGTERM: 15,
+  };
+
+  Object.keys(signals).forEach((signal) => {
+    process.on(signal, async () => {
+      await app.close();
+
+      const exitSignal = signals[signal] + 128;
```
Comment (jrhender) [inline authorization-server/src/main.ts:40]: It would be good to explain where `128` is coming from here

**Review by jrhender — APPROVED** [review]
Looks good, just one minor comment

**Pull request merged**